### PR TITLE
[in_app_purchases]specify cannot test on ios simulator.

### PR DESCRIPTION
**Pull request by Sameerkash**
Adds API docs and readme to tell the user that In-app Purchases cannot be tested on a simulator as StoreKit API is not supported on the simulator 

closes flutter/flutter#74181

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test exempt.
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [ x I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart [hunk 1]
```diff
@@ -90,6 +90,11 @@ abstract class InAppPurchaseConnection {
   ///
   /// See the [example readme](../../../../example/README.md) for steps on how
   /// to initialize products on both payment platforms.
+  ///
```
Comment (cyanglaz) [inline packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart:93]: Is this the best place for this doc? I'd imaging most of the APIs won't work. I think specifying this in README is good enough.

**Review by cyanglaz — COMMENTED** [review]
I think testing on simulator is possible now https://developer.apple.com/documentation/storekit/in-app_purchase/testing_in-app_purchases_in_xcode?language=objc

File: packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart (hunk 1 above)
Comment (Sameerkash) [reply packages/in_app_purchase/lib/src/in_app_purchase/in_app_purchase_connection.dart:93]: OKay will remove that.

**Comment by Sameerkash** [comment]
@cyanglaz 

> I think testing on the simulator is possible now https://developer.apple.com/documentation/storekit/in-app_purchase/testing_in-app_purchases_in_xcode?language=objc

I felt this doc is ambiguous cause I did a lot of research went through several StackOverflow posts and the majority stated that this wasn't possible and the answers were as recent as late 2020. Hence I landed on the conclusion. 

Although one of the threads mentioned there were some bugs and that's why it couldn't be tested, but nowhere on the apple docs its clear I guess. 

If you can confirm it is possible, we can close this.

**Comment by cyanglaz** [comment]
@Sameerkash Yeah it is pretty random. i remember it used to be testable then not. As it for today, Apple says it is testable on their doc that I linked. So I believe it should be. But I agree, we should confirm it is testable before closing this issue/PR.

**Comment by stuartmorgan** [comment]
Based on the discussion in the SO question linked from the issue, it does not appear that this is a hard restriction. Given that, and that the docs from Apple linked above specifically say that it can run on simulator or device, this text doesn't seem to be correct, so I'm going to close this. If at some point there is definitive evidence that it cannot ever be run on simulator, we can re-open this.

**Comment by cyanglaz** [comment]
Confirmed that testing on simulator is possible following the new iOS 14 store kit testing steps here: https://developer.apple.com/documentation/xcode/setting_up_storekit_testing_in_xcode?language=objc#3625696

**Comment by pradeep-orbi** [comment]
> https://developer.apple.com/documentation/xcode/setting_up_storekit_testing_in_xcode?language=objc#3625696

This is to test with the data that is configured locally. How to test with the consumables from the app store connect using simulator ?

**Comment by kpradeepkumarreddy** [comment]
> Confirmed that testing on simulator is possible following the new iOS 14 store kit testing steps here: https://developer.apple.com/documentation/xcode/setting_up_storekit_testing_in_xcode?language=objc#3625696

This is to test with the data that is configured locally. How to test with the consumables from the app store connect using simulator ?

**Comment by cyanglaz** [comment]
@kpradeepkumarreddy There's no way to test real IAP data from app store with simulators.

**Pull request closed without merging**